log4j-1.2-api/src/main/java/org/apache/log4j/spi/ThrowableInformation.java

```diff
@@ -29,8 +29,8 @@
 
     static final long serialVersionUID = -4748765566864322735L;
 
-    private transient Throwable throwable;
-    private Method toStringList;
+    private final transient Throwable throwable;
```

Comment:
Can you please elaborate? I don't see reasons to not do them `final`

Reply:
final fields are either initialized in the constructor or as assignments on their declaration. Many fields that are declared transient because the values assigned to them are not serializable, not because it is OK for them to be null when the object is reinstantiated via deserialization. It is quite possible that the objects can be recreated during deserialization from the data that is present. But if the fields are marked final those recreated objects cannot be stored in the appropriate fields without using reflection. 

In summary, marking a field as transient simply means it cannot be automatically serialized and deserialized. In no way does it imply that is expected to always be null.

Reply:
>It is quite possible that the objects can be recreated during deserialization from the data that is present.

But there is no such code. It means they are not recreated. I someone in future will deserialization code - he can also remove `final` modifier.

Moreover if classes `FormattedMessage`, `StringFormattedMessage` and `ParameterizedMessage`: code is already ready to have this fields `final` and `null`. Such fields are not serialized and not read from ObjectOutputStream.

If you insist - I will remove final

Reply:
Just couple examples to show, that having `final transient` it's ok:
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/time/ZoneRegion.java#L104
https://github.com/OpenGamma/Strata/blob/master/modules/collect/src/main/java/com/opengamma/strata/collect/array/DoubleMatrix.java#L66
https://github.com/google/guava/blob/master/guava/src/com/google/common/reflect/TypeToken.java#L701